File: Sieve-of-Eratosthenes.py
#!/usr/bin/python

from array import array

CONST_N = 200

primes = array("I")
is_prime = [] # Couldn't use array(), because the array() doesn't has constructor that accept `bool`

for i in range(0, CONST_N):
	primes.append(i)
	is_prime.append(True)

for i in range(2, CONST_N):
	if is_prime[i]:
		for j in range(i * i, CONST_N, i):
			is_prime[j] = False

for i in range(2, CONST_N):
	if is_prime[i]:
		print(primes[i], end='\n')

File: Sieve-of-Eratosthenes.rb
#!/usr/bin/ruby

CONST_N = 200

primes = Array.new
is_prime = Array.new(200, true)

for i in 0...200
	primes[i] = i;
    is_prime[i] = true;
end

for i in 2...200
    if is_prime[i]
        (i * i...CONST_N).step(i).each do |j|
            is_prime[j] = false
        end
    end
end

for i in 2...CONST_N
    if is_prime[i]
        puts(primes[i])
    end
end
